lib/solidus_importer/processors/option_values.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module SolidusImporter
+  module Processors
+    class OptionValues < Base
+      def call(context)
+        @data = context.fetch(:data)
+        return unless option_value_names.any?
+
+        variant = context.fetch(:variant)
+        process_option_values(variant)
+      end
+
+      private
+
+      def process_option_values(variant)
+        option_value_names.each_with_index do |name, i|
+          option_type = option_type(variant.product, i)
+
+          v = Spree::OptionValue.find_or_initialize_by(
+            option_type: option_type,
+            name: name
+          )
+          v.presentation = name
+          variant.option_values << v
+        end
+      end
+
+      def option_type(product, index)
+        product.option_types.find_by(position: index + 1)
```

Comment:
A way to skip the `find_by` here could be to attach the option types to the context in OptionTypes processor.

Reply:
I totally agree! 👍

Reply:
I amend previous comment: this solution doesn't fit since each processor works on a row, so we have no reference to the product's option types because they're not replicated on every row, but they're defined in the first row only.